### PR TITLE
curvefs/client: add parser for --mdsAddr

### PR DESCRIPTION
**Pull request by h0hmj**
### What problem does this PR solve?

Issue Number: #1422 

### What is changed and how it works?

What's Changed:

parse args and update argc/argv before pass to fuse_opt_parse 

How it Works:

Side effects(Breaking backward compatibility? Performance regression?):

### Check List

- [ ] Relevant documentation/comments is changed or added
- [ ] I acknowledge that all my contributions will be made under the project's license

### CODE REVIEW

File: curvefs/src/client/main.c [hunk 1]
```diff
@@ -64,23 +64,50 @@ void extra_options_help() {
...
     print_option_help("conf", "[required] path of config file");
-    print_option_help("mdsAddr", "mdsAddr of curvefs cluster");
+    printf("    --mdsAddr              mdsAddr of curvefs cluster\n");
+}
+
+char* parse_option_by_pattern(int* argc, char*** argv, const char* pattern) {
+    // parse option that fuse_opt_parse cannot parse
+    // remove it from argv, update argc
+    int newArgc = *argc;
+    char** newArgv = (char**)malloc(sizeof(char*) * newArgc);
```
Comment (wu-hanqing) [inline curvefs/src/client/main.c:74]: this memory is leaked, but it's acceptable.

File: curvefs/src/client/main.c [hunk 2]
```diff
@@ -64,23 +64,50 @@ void extra_options_help() {
...
+    char* out = NULL;
+    for (int i = 0, j = 0; j < *argc; j++) {
+        int len1 = strlen((*argv)[j]);
+        int len2 = strlen(pattern);
```
Comment (wu-hanqing) [inline curvefs/src/client/main.c:78]: calculate it once at the beginning is better

File: curvefs/src/client/main.c [hunk 3]
```diff
@@ -64,23 +64,50 @@ void extra_options_help() {
...
-    struct MountOption mOpts = {0};
     int ret = -1;
 
     if (fuse_parse_cmdline(&args, &opts) != 0)
         return 1;
     if (opts.show_help) {
         printf(
             "usage: %s -o conf=/etc/curvefs/client.conf -o fsname=testfs \\\n"
-            "       -o fstype=s3 [-o mdsaddr=1.1.1.1] [OPTIONS] <mountpoint>\n",
+            "       -o fstype=s3 [--mdsaddr=1.1.1.1,2.2.2.2] [OPTIONS] <mountpoint>\n",
```
Comment (wu-hanqing) [inline curvefs/src/client/main.c:110]: what about the port?

**Review by SeanHai — APPROVED** [review]

File: curvefs/src/client/main.cpp [hunk 4]
```diff
@@ -21,37 +21,38 @@
  * Author: xuchaojie
  */
 
+#include <unordered_map>
 #include "curvefs/src/client/curve_fuse_op.h"
 #include "curvefs/src/client/fuse_common.h"
 
 static const struct fuse_lowlevel_ops curve_ll_oper = {
```
Comment (h0hmj) [inline curvefs/src/client/main.cpp:28]: reorder it to match c struct

File: curvefs/src/client/main.c (hunk 3 above)
Comment (h0hmj) [reply curvefs/src/client/main.c]: add port to example

File: curvefs/src/client/main.cpp (hunk 4 above)
Comment (h0hmj) [reply curvefs/src/client/main.cpp:28]: add fuse version limit and fill all fields

File: curvefs/src/client/main.c (hunk 1 above)
Comment (h0hmj) [reply curvefs/src/client/main.c]: rewrite

**Comment by h0hmj** [comment]
recheck

**Review by wu-hanqing — APPROVED** [review]

**Pull request merged**